numba_dpex/dpnp_iface/_intrinsic.py

```diff
@@ -147,6 +158,14 @@ def _get_queue_ref(context, builder, sig, args):
     return ret
 
 
+def _update_queue_attr(array, queue):
+    """Sets the sycl_queue member of an ArrayStruct."""
```

Comment:
This explanation is not clear. Looks like it's been called when we are assigning a queue object into an usmndarray.

Reply:
How about:

    Assigns the sycl_queue member of an usmarystruct_t instance.

    After creating a new usmarystruct_t struct (e.g. in _empty_nd_impl) the
    members of the struct are populated by calling
    numba.np.arrayobj.populate_array. The populate_array function does not
    update the sycl_queue member as populate_array is written specifically for
    numba's arystruct_t type that does not have a sycl_queue member. The
    _update_queue_attr is a helper function to update the sycl_queue field.